### ISSUE TITLE
Style/HashAsLastArrayItem ignores arrays with a single hash element

### ISSUE THREAD

**Issue by phil-workato** [rubocop/rubocop#12027]
```ruby
[foo: 1] # => [{:foo => 1}]
```

## Expected behavior

```
Offenses:

1.rb:3:6: C: [Correctable] Style/HashAsLastArrayItem: Wrap hash in { and }.
x = [foo: 1]
     ^^^^^^
```

## Actual behavior

No offence.

## Steps to reproduce the problem

```ruby
x = [1, foo: 1]
```

## RuboCop version

```
$ rubocop -V
1.41.1 (using Parser 3.2.2.3, rubocop-ast 1.24.0, running on ruby 2.7.7) [arm64-darwin22]
  - rubocop-rails 2.20.2
  - rubocop-rspec 2.22.0
  - rubocop-sorbet 0.7.0
```

**Comment by koic** [comment]
According to the NOTE of `Style/HashAsLastArrayItem` cop this looks like expected behavior:

> This cop will ignore arrays where all items are hashes, regardless of EnforcedStyle.

https://docs.rubocop.org/rubocop/1.54/cops_style.html#stylehashaslastarrayitem

**Comment by pirj** [comment]
@koic there might have been a misunderstanding.
‘[1, foo: 1]’ doesn’t fall under the “arrays where all items are hashes”, does it?

**Comment by koic** [comment]
@pirj That case looks detectable :-)


```console
$ echo 'x = [1, foo: 1]' | be rubocop --stdin example.rb -a --only Style/HashAsLastArrayItem
Inspecting 1 file
C

Offenses:

example.rb:1:9: C: [Corrected] Style/HashAsLastArrayItem: Wrap hash in { and }.
x = [1, foo: 1]
        ^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
====================
x = [1, {foo: 1}]
```
